3-UsingAPIs/Week1/homework/ex1-johnWho.js

```diff
@@ -7,20 +7,25 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
   didn't pass in a first name!"
 ------------------------------------------------------------------------------*/
 // TODO see above
-const getAnonName = (firstName, callback) => {
-  setTimeout(() => {
-    if (!firstName) {
-      callback(new Error("You didn't pass in a first name!"));
-      return;
-    }
+function getAnonName(firstName) {
+  const fullName = `${firstName} Doe`;
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      if (firstName) {
+        resolve(fullName);
+      } else {
+        reject(new Error("You didn't pass in a first name!"));
+      }
+    }, 1000);
+  });
+}
 
-    const fullName = `${firstName} Doe`;
-
-    callback(fullName);
-  }, 1000);
-};
-
-getAnonName('John', console.log);
+getAnonName('john')
+  .then((fullName) => {
+    console.log(fullName);
+    return getAnonName();
+  })
```

Comment:
Looks good but you can simplify this like so:
```suggestion
  .then(console.log)
```

Reply:
ok!